app.test.js

```diff
@@ -108,12 +107,130 @@ describe("GET: /api/articles", () => {
+    test("Return new comment to specified article", () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "icellusedkars"})
+      .expect(201)
+      .then(({body}) => {
+        expect((expect((body.body)).toBe("Hello")))
+        expect((body.author)).toBe("icellusedkars")
+      })
+    })
+    test("Return not found if id is incorrect", () => {
+      return request(app)
+      .post('/api/articles/3945/comments')
+      .send({body: "Hello", username: "icellusedkars"})
+      .expect(400)
+      .then(({body}) => {
+      expect(body.msg).toBe('not an id')
+    })
+    })
+    test("Ensure unnessiary properties are ignored", () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "icellusedkars", chicken: "waffle"})
+      .expect(201)
+      .then(({body}) => {
+        expect(body).toMatchObject( {
+          comment_id: 19,
+          body: 'Hello',
+          article_id: 9,
+          author: 'icellusedkars',
+          votes: 0
+        })
+        expect(body.chicken).toBe(undefined)
+      })
+    })
+    test('Return 404 if username invalid', () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "Marlon-Brando"})
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe('Username doesnt exist')
+      })
+    })
+  })
+})
+describe("PATCH: /api/articles/:article_id", () => {
+  test('Return updated article votes', () => {
+    return request(app)
+          .patch('/api/articles/1')
+          .send({inc_votes : -33})
+          .expect(200)
+          .then((data) => {
+            expect(data.body).toMatchObject({
+              article_id: 1,
+              title: 'Living in the shadow of a great man',
+              topic: 'mitch',
+              author: 'butter_bridge',
+              body: 'I find this existence challenging',
+              votes: 67,
+              article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+            })
+          })
+        })
+  test('Return 404 if article ID invalid', () => {
+      return request(app)
+        .patch('/api/articles/93939')
+        .send({inc_votes : -33})
+        .expect(404)
+        .then(({body}) => {
+          expect(body.msg).toBe('Article not found')
+            })
+          })
+
+
+})
+
+describe("DELETE: /api/comments/:comment_id", () => {
+  test("delete the given comment by comment_id", () => {
+    return request(app)
+    .delete("/api/comments/1")
+    .expect(204)
+    .then((body) => {
+      expect(body.body).toEqual({})
+    })
+  })
+  test("return 404 comment not found if given invalid ID", () => {
+    return request(app)
+    .delete("/api/comments/38585")
+    .expect(404)
+    .then(({body}) => {
+      expect(body.msg).toEqual('Comment not found')
+    })
+  })
```

Comment:
Nice work here! I'd consider adding a test for: **Status 400, invalid ID, e.g "not-an-id"**